src/mono/mono/metadata/native-library.c

```diff
@@ -1240,8 +1299,20 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 #ifndef HOST_WIN32
 retry_with_libcoreclr:
 #endif
-	// FIXME: these flags are not getting passed correctly
-	module = netcore_lookup_native_library (alc, image, new_scope, 0);
+	cinfo = mono_custom_attrs_from_method_checked (method, error);
+	flags = get_dllimportsearchpath_flags (cinfo);
+	if (cinfo && !cinfo->cached)
+		mono_custom_attrs_free (cinfo);
+
+	if (flags < 0) {
+		cinfo = mono_custom_attrs_from_assembly_checked (m_class_get_image (method->klass)->assembly, TRUE, error);
```

Comment:
In both of the `_checked` calls, you need to actually look at `error` and do something with it. Probably ok to just `mono_error_cleanup (error)` .  But in that case you should actually just use 2 separete local errors like this:
```
		{
				ERROR_DECL (local_error);
				cinfo = mono_custom_attrs_from_method_checked (method, local_error);
				mono_error_cleanup (local_error);
		}
		flags = get_dllimportsearchpath_flags (cinfo);
		... /* etc */
```